google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -201,6 +209,8 @@ public void removeListener(Runnable listener) {
 
     volatile ByteString transactionId;
 
+    ByteString previousTransactionId;
```

Comment:
nit: can this be `final`?

Reply:
This field is modified in unit tests, so it cannot be declared as final

Reply:
Fair point. But those unit tests are both mocking `TransactionContextImpl`. So could we then solve that like this:
1. Make this field final.
2. Add a package-private `getPreviousTransactionId()` that returns this field. Use this getter everywhere in this class where the field would otherwise be accessed.
3. Return a mocked result from that getter in the unit tests that need to have it return something else.

Reply:
Updated `previousTransactionId` field to be `final`.